### ISSUE TITLE
Worker task remain after connector removal

### ISSUE THREAD

**Issue by abraham-leal** [snowflakedb/snowflake-kafka-connector#112]
`[2020-03-23 00:08:27,731] INFO Successfully processed removal of connector 'snow_sink' (org.apache.kafka.connect.storage.KafkaConfigBackingStore)
[2020-03-23 00:08:27,731] INFO [Worker clientId=connect-1, groupId=connect-cluster] Connector snow_sink config removed (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
[2020-03-23 00:08:27,731] INFO [Worker clientId=connect-1, groupId=connect-cluster] Handling connector-only config update by stopping connector snow_sink (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
[2020-03-23 00:08:27,731] INFO Stopping connector snow_sink (org.apache.kafka.connect.runtime.Worker)
[SF_KAFKA_CONNECTOR] SnowflakeSinkConnector:stop (com.snowflake.kafka.connector.SnowflakeSinkConnector)
[2020-03-23 00:08:27,732] INFO Stopped connector snow_sink (org.apache.kafka.connect.runtime.Worker)
[2020-03-23 00:08:28,474] INFO [Worker clientId=connect-1, groupId=connect-cluster] Joined group at generation 216 with protocol version 2 and got assignment: Assignment{error=0, leader='connect-1-7e5cc2d8-a54a-4b0a-95a2-b245b10fce66', leaderUrl='http://ec2-34-213-4-7.us-west-2.compute.amazonaws.com:8083/', offset=304, connectorIds=[], taskIds=[], revokedConnectorIds=[snow_sink], revokedTaskIds=[snow_sink-0, snow_sink-1, snow_sink-3], delay=5179} with rebalance delay: 5179 (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
[SF_KAFKA_CONNECTOR] SnowflakeSinkTask[ID:0]:put 0 records (com.snowflake.kafka.connector.SnowflakeSinkTask)
[SF_KAFKA_CONNECTOR] SnowflakeSinkTask[ID:3]:put 0 records (com.snowflake.kafka.connector.SnowflakeSinkTask)
[SF_KAFKA_CONNECTOR] SnowflakeSinkTask[ID:1]:put 0 records (com.snowflake.kafka.connector.SnowflakeSinkTask)
[SF_KAFKA_CONNECTOR] SnowflakeSinkTask[ID:0]:put 0 records (com.snowflake.kafka.connector.SnowflakeSinkTask)
[SF_KAFKA_CONNECTOR] SnowflakeSinkTask[ID:3]:put 0 records (com.snowflake.kafka.connector.SnowflakeSinkTask)
`

Deleting the connector doesn't seem to revoke the tasks correctly. Please see log above.

**Comment by sfc-gh-zli** [comment]
How did you close the connector? If by sending DELETE request to REST api of Kafka Connect, this shouldn't happen.

**Comment by seruman** [comment]
We also experienced similar issue. After deleting  connnector via Kafka Connect's REST API, there were clients in the consumer group even after minutes, and JMX metrics for connector was still available. We had to stop Kafka Connect process.

We've checked connected clients with Kafka's shell tools;

```
kafka-consumer-groups.sh --bootstrap-server $BS --describe --group $CG
```

Sorry that I can not provide logs since we have multiple Snowflake Kafka Connectors running and not able to distinguish their logs. https://github.com/snowflakedb/snowflake-kafka-connector/issues/87#issuecomment-644921374

**Comment by sfc-gh-zli** [comment]
I couldn't reproduce this error on my side. We have test that close and reopen the connector for three times and I didn't see any thread leaked. SnowflakeSinkTask is closed properly as well. 
Maybe that is caused by thread leaks in version < 1.3.

**Comment by abraham-leal** [comment]
@sfc-gh-zli Yup, this was reported in v1.2.1, my guess is this leak got fixed by #166. Closing this issue, thanks.
